### DISCUSSION TITLE
[Question] Is plugin(...) not necesarry?

### DISCUSSION THREAD

**Discussion by zaydek** [tailwindlabs/tailwindcss#1428] [General]
I’ve been pretty confused looking at other people’s source code and comparing that to the official docs, which states:

> To get started with your first plugin, import Tailwind's `plugin` function from `tailwindcss/plugin`, and call it with an anonymous function as the first argument in the `plugins` list in your config file:
>
> https://tailwindcss.com/docs/plugins/#app

I have found that you can _omit_ the call to `plugin` and everything _appears_ to work as expected — does anyone understand why that works?

**Comment by tlgreg** [comment]
Yes, tailwind accepts only the handler function as the plugin. Originally, that was the only option.

`1.2` added the option for plugins to extend the config and with that, the `tailwindcss/plugin` function was also added.

In the future, other features might get added to plugins through the function or it could become mandatory (in a major release).

**Comment by zaydek** [comment]
Ah. I found this super confusing as the docs demonstrate creating a plugin by using `plugin`. Anyway, I’ve got it working now. Thank you.

**Comment by adamwathan** [comment]
Don't worry about what you see in other repos, just worry about what it says in the docs :) What's in the docs works and is what we recommend.

**Comment by zaydek** [comment]
Thank you!
